packages/modules/gcloud/src/cloudstorage-emulator-container.ts

```diff
@@ -0,0 +1,102 @@
+import { AbstractStartedContainer, GenericContainer, Wait } from "testcontainers";
+import type { InspectResult, StartedTestContainer } from "testcontainers";
+
+const PORT = 4443;
+const DEFAULT_IMAGE = "fsouza/fake-gcs-server";
+
+export class CloudStorageEmulatorContainer extends GenericContainer {
+  private _externalURL?: string;
+
+  constructor(image = DEFAULT_IMAGE) {
+    super(image);
+
+    this.withExposedPorts(PORT)
+      .withEntrypoint([
+        "fake-gcs-server",
+        "-scheme",
+        "http",
+        ...(this._externalURL ? ["-external-url", this._externalURL] : []),
+      ])
+      .withWaitStrategy(Wait.forLogMessage(/server started/g, 1))
+      .withStartupTimeout(120_000);
+  }
+
+  public withExternalURL(url: string): CloudStorageEmulatorContainer {
+    this._externalURL = url;
```

Comment:
The external URL is only used in the constructor, so this doesn't currently do anything. You should probably update the entrypoint on `start` seeing that the command can change. Is it possible to add a test for this?